### DISCUSSION TITLE
Create new HUD Item

### DISCUSSION THREAD

**Discussion by Tiebe** [FabricMC/fabric-api#1611] [Mod Dev Support] [answered]
I'm trying to draw an image on screen, as a new HUD item. I've tried this:
```
        RenderSystem.setShader(GameRenderer::getPositionTexShader);
        RenderSystem.setShaderColor(1.0F, 1.0F, 1.0F, 1.0F);
        RenderSystem.setShaderTexture(0, image);
        MatrixStack matrices = new MatrixStack();
        Screen.drawTexture(matrices, 0, 0, 0, 0, width, height, width, height);
```

But it seems to draw it with transparency, so how do I draw it without transparency?

The image drawn on screen:

![](https://cdn.discordapp.com/attachments/507982478276034570/871844859873476658/unknown.png)

The image it's supposed to draw:

![](https://cdn.discordapp.com/attachments/507982478276034570/871845356781060136/image.png)

**Comment by warjort** [comment, marked as answer]
Are you using the fabric event?
````
HudRenderCallback.EVENT.register((maxtrixStack, tickDelta) -> { 
/* Your code here */
});
````
I believe the rendering system is supposed to be pretty much in the default state when it is called?

Anyway, try something like:
````
RenderSystem.disableBlend();
RenderSystem.defaultBlendFunc();
````
But you might break some other rendering if you do this at a "random" point in the hud rendering code.

You can also look at Minecraft's RenderPhase and RenderLayer classes to see some common use cases of the RenderSystem.

**Reply by Tiebe** [reply]
Thanks, I did `disableBlend()` at the beginning and `enableBlend()` at the end and it works perfectly now!
